libraries/abstractions/common_io/include/iot_i2c.h

```diff
@@ -215,52 +217,102 @@ int32_t iot_i2c_read_sync( IotI2CHandle_t const pxI2CPeripheral,
  * @param[in] xBytes The number of bytes to write.
  *
  * @return
- * - IOT_I2C_SUCCESS, on success and all the requested bytes have been written
+ * - IOT_I2C_SUCCESS, on success
  * - IOT_I2C_INVALID_VALUE, if
  *     - pxI2CPeripheral is NULL
  *     - pxI2CPeripheral is not opened yet
  *     - pucBuffer is NULL
  *     - xBytes is 0
- * - IOT_I2C_NACK, if
+ * - IOT_I2C_SLAVE_ADDRESS_NOT_SET, if slave address is not set yet
+ * - IOT_I2C_NACK or IOT_I2C_WRITE_FAILED, if
  *     - no device correspond to the slave address
  *     - the slave is unable to receive or transmit
  *     - the slave gets data or commands that it does not understand
- *     - not all the requested data is available.
- * - IOT_I2C_BUS_TIMEOUT, if this is a SMBus and slave device does not respond within configured timeout.
+ *     - there is some unknown driver error
+ * - IOT_I2C_BUS_TIMEOUT, if timeout is supported and slave device does not respond within configured timeout.
  * - IOT_I2C_BUSY, if the bus is busy which means there is an ongoing transaction.
- * - IOT_I2C_WRITE_FAILED, if there is some unknown driver error
+ * - IOT_I2C_FUNCTION_NOT_SUPPORTED, if this board doesn't support this operation.
  */
 int32_t iot_i2c_write_sync( IotI2CHandle_t const pxI2CPeripheral,
                             uint8_t * const pucBuffer,
                             size_t xBytes );
 
 /**
- * @brief Starts the I2C master read operation in non-blocking mode.
- *        You write_sync the address of the register you want to read from first to read data
+ * @brief Starts the I2C master read operation in asynchronous mode.
+ *
+ * This function attempts to read certain number of bytes from a pre-allocated buffer, in asynchronous way.
+ * It returns immediately when the operation is started and the status can be check by calling iot_i2c_ioctl.
+ * Once the operation completes, successful or not, the user callback will be invoked.
+ *
+ * Partial read might happen, e.g. slave device unable to receive more data.
+ * And the number of bytes that have been actually read can be obtained by calling iot_i2c_ioctl.
+ *
+ * @Note Usually, the address of register needs to be written before calling this function.
```

Comment:
Suggest adding comment about usually calling iot_i2c_set_callback() prior to this, so that implementor can get notified when the async call is complete.

Reply:
Yes, will do.